LeaderboardBackend/Controllers/AccountController.cs

```diff
@@ -74,4 +72,59 @@ public async Task<ActionResult<UserViewModel>> Register([FromBody] RegisterReque
                 return Conflict(new ValidationProblemDetails(ModelState));
             });
     }
+
+    /// <summary>
+    ///     Logs a User in.
+    /// </summary>
+    /// <param name="request">
+    ///     The `LoginRequest` instance from which to perform the login.
+    /// </param>
+    /// <param name="authService">IAuthService injection.</param>
+    /// <response code="200">
+    ///     The `User` was logged in successfully. A `LoginResponse` is returned, containing a token.
+    /// </response>
+    /// <response code="400">The request was malformed.</response>
+    /// <response code="401">The password given was incorrect.</response>
+    /// <response code="403">The associated `User` is banned.</response>
+    /// <response code="404">No `User` with the requested details could be found.</response>
+    /// <response code="422">
+    ///     The request contains errors.<br/><br/>
+    ///     Validation error codes by property:
+    ///     - **Password**:
+    ///       - **NotEmptyValidator**: No password was passed
+    ///       - **PasswordFormat**: Invalid password format
+    ///     - **Email**:
+    ///       - **NotNullValidator**: No email was passed
+    ///       - **EmailValidator**: Invalid email format
+    /// </response>
+    [AllowAnonymous]
+    [HttpPost("/login")]
+    [ApiConventionMethod(typeof(Conventions), nameof(Conventions.PostAnon))]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<LoginResponse>> Login([FromBody] LoginRequest request, [FromServices] IAuthService authService)
+    {
+        LoginResult result = await _userService.LoginByEmailAndPassword(request.Email, request.Password);
+
+        return result.Match<ActionResult<LoginResponse>>(
+            loginToken => Ok(new LoginResponse { Token = loginToken }),
+            errors =>
+            {
+                if (errors.NotFound)
+                {
+                    return NotFound();
+                }
+
+                if (errors.Banned)
+                {
+                    return Forbid();
+                }
+
+                // errors.WrongPassword implicitly true
+                return Unauthorized();
```

Comment:
I don't like what I've done here, but I think the alternatives are worse, so 🤷

Reply:
You could use [pattern matching](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#multiple-inputs).

Reply:
Discussed it with @Dalet and I just need to leverage the `OneOf` library better. I'll push a commit soon showing what I mean.

Reply:
https://github.com/leaderboardsgg/leaderboard-backend/pull/169/commits/53064f1e1ec758f7f5d4fe45415e51130d7fc492